frontend/components/TableContainer/TableContainer.tsx

```diff
@@ -223,10 +223,10 @@ const TableContainer = <T,>({
 
   // NOTE: used to reset page number to 0 when modifying filters
   useEffect(() => {
-    if (pageIndex !== 0 && resetPageIndex && !isClientSidePagination) {
+    if (resetPageIndex && !isClientSidePagination) {
       onPaginationChange(0);
     }
-  }, [resetPageIndex, pageIndex, isClientSidePagination]);
```

Comment:
`resetPageIndex` being `true` and this effect's dependence on `pageIndex` caused the bug. Changing the team resulted in `resetPageIndex` being `true`, and subsequently trying to navigate to any `page !==0` led to an immediate reset of the page to 0.